pkg/microservice/aslan/core/system/service/registry.go

```diff
@@ -199,36 +206,100 @@ func ListAllRepos(log *zap.SugaredLogger) ([]*RepoInfo, error) {
+}
+
 func ListReposTags(registryInfo *commonmodels.RegistryNamespace, names []string, logger *zap.SugaredLogger) ([]*RepoImgResp, error) {
 	var regService registry.Service
 	if registryInfo.AdvancedSetting != nil {
 		regService = registry.NewV2Service(registryInfo.RegProvider, registryInfo.AdvancedSetting.TLSEnabled, registryInfo.AdvancedSetting.TLSCert)
 	} else {
 		regService = registry.NewV2Service(registryInfo.RegProvider, true, "")
 	}
+
+	endPoint := registry.Endpoint{
+		Addr:      registryInfo.RegAddr,
+		Ak:        registryInfo.AccessKey,
+		Sk:        registryInfo.SecretKey,
+		Namespace: registryInfo.Namespace,
+		Region:    registryInfo.Region,
+	}
 	repos, err := regService.ListRepoImages(registry.ListRepoImagesOption{
-		Endpoint: registry.Endpoint{
-			Addr:      registryInfo.RegAddr,
-			Ak:        registryInfo.AccessKey,
-			Sk:        registryInfo.SecretKey,
-			Namespace: registryInfo.Namespace,
-			Region:    registryInfo.Region,
-		},
-		Repos: names,
+		Endpoint: endPoint,
+		Repos:    names,
 	}, logger)
 
 	images := make([]*RepoImgResp, 0)
 	if err == nil {
 		for _, repo := range repos.Repos {
+			// find all tags from image tags db
+			imageTags, err := commonrepo.NewImageTagsCollColl().Find(&commonrepo.ImageTagsFindOption{
+				RegistryID:  registryInfo.ID.Hex(),
+				RegProvider: registryInfo.RegProvider,
+				ImageName:   repo.Name,
+				Namespace:   registryInfo.Namespace,
+			})
+			dbTags := make(map[string]*commonmodels.ImageTag, 0)
+			if err != nil {
+				if err != mongo.ErrNoDocuments && err != mongo.ErrNilDocument {
+					logger.Errorf("find tags of the image %s from db error: %v", repo.Name, err)
+					return nil, err
+				} else {
+					logger.Errorf("find 0 tags of the image %s from db, the error is: %v", repo.Name, err)
+				}
+			} else {
+				for _, imageTag := range imageTags.ImageTags {
+					dbTags[imageTag.TagName] = imageTag
+				}
+			}
+
+			// get the difference between the tags from image tags db and the tags from registry
+			newTags := &newTags{
+				Resp: make([]*RepoImgResp, 0),
+				m:    &sync.Mutex{},
+			}
+			wg := &sync.WaitGroup{}
+			count := 0
+			logger.Infof("get image[%s] %d tags from db, get image[%s] %d tags from registry", repo.Name, len(dbTags), repo.Name, len(repo.Tags))
 			for _, tag := range repo.Tags {
-				img := &RepoImgResp{
-					Host:  util.TrimURLScheme(registryInfo.RegAddr),
-					Owner: repo.Namespace,
-					Name:  repo.Name,
-					Tag:   tag,
+				if t, ok := dbTags[tag]; ok {
+					resp := &RepoImgResp{
+						Host:       util.TrimURLScheme(registryInfo.RegAddr),
+						Owner:      repo.Namespace,
+						Name:       repo.Name,
+						Tag:        t.TagName,
+						created:    t.Created,
+						FormatTime: time.Unix(t.Created, 0).Format("2006-01-02 15:04:05"),
+					}
+					images = append(images, resp)
+
+					continue
 				}
-				images = append(images, img)
+
+				count++
+				if count%100 == 0 {
+					time.Sleep(time.Millisecond * 100)
+				}
```

Comment:
it's not a graceful way to control the groutine concurrency, what if the 'get image detail' request costs more than 100ms? This code needs optimzied, consider use cosumer model, take go routine as a cosumer, and the amount of goroutines can be set exactly.